lib/services/protocol_service.ts

```diff
@@ -522,6 +522,13 @@ export class SNProtocolService extends PureService implements EncryptionDelegate
     if (format === PayloadFormat.DecryptedBareObject) {
       return payload;
     }
+    /** We check if the key can decrypt the encrypted payload. */
+    if (
+      payload.usesItemKeyEncryption() &&
+      key?.content_type === ContentType.RootKey
+    ) {
+      key = undefined;
```

Comment:
I think we should throw an exception here. We shouldn't be making up for a caller's misuse of our function. If you pass a payload and the wrong key type, this is a logical error and an exception should be thrown. So we should solve this one level up in the stack trace.